### PR TITLE
RSDK-1239 - properly handle component with geometry and model json

### PR DESCRIPTION
**Pull request by biotinker**

### CODE REVIEW

File: referenceframe/transformable.go [hunk 1]
```diff
@@ -205,6 +205,9 @@ func (gF *GeometriesInFrame) Parent() string {
 
 // Geometries returns the geometries observed.
 func (gF *GeometriesInFrame) Geometries() map[string]spatialmath.Geometry {
+	if gF.geometries == nil {
+		return make(map[string]spatialmath.Geometry)
```
Comment (spieswl) [inline referenceframe/transformable.go:209]: Is there a reason we would not want to assign the newly allocated map to `gF.geometries` and let the function fall to the unconditional return statement?

**Review by spieswl — APPROVED** [review]
I think my comment is more of a stylistic clarification, the rest of the PR looks good.

File: referenceframe/transformable.go (hunk 1 above)
Comment (biotinker) [reply referenceframe/transformable.go:209]: I tend to err against silently editing data structures that the user initialized

Comment (biotinker) [reply referenceframe/transformable.go:209]: This allows other, future functionality to also be able to distinguish between "this was initialized with nil" and "this was initialized with a zero-length map"

**Pull request merged**